src/classes/ADDR_AdminAcc_TEST.cls

```diff
@@ -0,0 +1,891 @@
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            addrs[i].Parent_Account__c = accsAddrs.accs[i].Id;
+            addrs[i].Default_Address__c = false;
+            addrs[i].MailingStreet__c = 'override' + i;
+            addrs[i].MailingCity__c = 'override' + i;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        insert addrs;
+        
+        // set the accounts' address overrides
+        for (Integer i = 0; i < accsAddrs.accs.size(); i++) {
+            Account acc = accsAddrs.accs[i];
+            acc.Current_Address__c = addrs[i].Id;
+            acc.is_Address_Override__c = true;
+        }
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        update accsAddrs.accs;        
+        
+        // now let's delete the override Addresses
+        ADDR_Addresses_TDTM.hasRunAddrTrigger = false;
+        Test.startTest();
+        delete addrs;
+        Test.stopTest();
+    
+        // verify that the Account address fields went back to the default
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
+        BillingPostalCode, BillingCountry, BillingLatitude, BillingLongitude, is_Address_Override__c, Current_Address__c from Account]);
+        
+        for (Account acc : mapAccIdAcc.values()) {
+            system.assert(acc.BillingStreet.contains('Street'));
+            system.assert(acc.BillingCity.contains('City')); 
+            system.assertEquals(false, acc.is_Address_Override__c);
+            system.assertNotEquals(null, acc.Current_Address__c);                
+        }
+    }
+
+    /*********************************************************************************************************
+    @description Making an Address non-default. That is, the address is no longer the default one for the Account.  
+    verify: no change to account address
+    **********************************************************************************************************/            
+    @isTest
+    public static void updateNonDefaultAddr() {
+        configSettings();
+        
+        // this creates a default Address for each Account
+        UTIL_UnitTestData_TEST.AccsWithAddrs accsAddrs = UTIL_UnitTestData_TEST.createTestAccsAddrs(2, UTIL_Describe.getAdminAccRecTypeID());
+        List<Address__c> addrs = accsAddrs.addrs;
+        
+        // now let's update the Addresses
+        for (Integer i = 0; i < addrs.size(); i++) {
+            Address__c addr = addrs[i];
+            addr.Default_Address__c = false;
+        }
+        system.assertEquals(false, ADDR_Addresses_TDTM.hasRunAddrTrigger);
+        Test.startTest();
+        update addrs;
+        Test.stopTest();
+    
+        // verify that the Account and Contacts don't share the same address and it's new for the Account!
+        map<Id, Account> mapAccIdAcc = new map<Id, Account>([select Id, Name, BillingStreet, BillingCity, BillingState, 
```

Comment:
If the table in this query contains more than 50'000 records a runtime exception will be thrown. Consider either introducing a `WHERE` clause, or adding a `LIMIT` to contain the number of records returned.